### ISSUE TITLE
CS, snapshot: FFI failure during package install on Windows

### ISSUE THREAD

**Issue by Bogdanp** [racket/racket#4795]
**What version of Racket are you using?**

CS snapshots from the snapshot sites.

**What program did you run?**

Utah: https://github.com/Bogdanp/setup-racket/actions/runs/6530768185/job/17730698004#step:11:6
Northwestern: https://github.com/Bogdanp/setup-racket/actions/runs/6530768185/job/17730692297#step:11:5

**What should have happened?**

The package should be installed successfully. It looks like this may be related to the recent FFI changes.

**If you got an error message, please include it here.**

```
Resolving "nevermore" via https://plt.cs.northwestern.edu/snapshots/20231013-feb3f1967c/catalog/
CertOpenSystemStoreW: implementation not found
  arguments...:
   "ROOT"
  context...:
   C:\Program Files\Racket-8.11.0.3\collects\ffi\unsafe\alloc.rkt:86:15
   C:\Program Files\Racket-8.11.0.3\collects\openssl\private\win32.rkt:52:0: load-win32-store
   C:\Program Files\Racket-8.11.0.3\collects\openssl\mzssl.rkt:692:0: ssl-load-default-verify-sources!
   C:\Program Files\Racket-8.11.0.3\collects\openssl\mzssl.rkt:855:0: secure-client-context!
   C:\Program Files\Racket-8.11.0.3\collects\openssl\mzssl.rkt:543:0: make-client-context*
   C:\Program Files\Racket-8.11.0.3\collects\openssl\mzssl.rkt:871:0: ssl-secure-client-context
   C:\Program Files\Racket-8.11.0.3\collects\openssl\mzssl.rkt:553:0: get-context
   C:\Program Files\Racket-8.11.0.3\collects\openssl\mzssl.rkt:379:18
   C:\Program Files\Racket-8.11.0.3\collects\ffi\unsafe\atomic.rkt:73:13
   C:\Program Files\Racket-8.11.0.3\collects\openssl\mzssl.rkt:1416:0: wrap-ports
   C:\Program Files\Racket-8.11.0.3\collects\racket\contract\private\arrow-val-first.rkt:555:3
   C:\Program Files\Racket-8.11.0.3\collects\net\http-client.rkt:67:0: http-conn-open!
   C:\Program Files\Racket-8.11.0.3\collects\net\http-client.rkt:274:0: http-conn-open
   C:\Program Files\Racket-8.11.0.3\collects\racket\contract\private\arrow-val-first.rkt:555:3
   C:\Program Files\Racket-8.11.0.3\collects\net\url.rkt:202:0: http://getpost-impure-port
   C:\Program Files\Racket-8.11.0.3\collects\net\url.rkt:305:0: get-pure-port/headers
   ...
```

**Please include any other relevant details**

Windows

**Comment by mflatt** [comment]
I have been struggling to replicate this, but finally it occurs to me that it's because I've been building on Windows. I finally tried installing a snapshot, and now I see a different but related error — so, probably it's an issue with cross compilation (as used to build snapshots).

**Comment by jbclements** [comment]
The windows bundles are all built using cross-compilation on unix hosts; would this bug affect the release? Or maybe the FFI changes aren’t part of the release?

John

**Comment by sorawee** [comment]
It's not a part of the release (the initial change is #4784 -- "Not to be merged until after the v8.11 branch.")

**Comment by jbclements** [comment]
okay, many thanks.
